agent/src/beerocks/slave/agent_ucc_listener.h

```diff
@@ -15,10 +15,15 @@
 #include <mutex>
 
 namespace beerocks {
+
+// Forward decleration for backhaul_manager context saving
+class backhaul_manager;
+
 class agent_ucc_listener : public beerocks_ucc_listener {
 public:
-    agent_ucc_listener(uint16_t port, const std::string &vendor, const std::string &model,
-                       const std::string &bridge_iface, SocketClient **controller_sd);
+    agent_ucc_listener(backhaul_manager &backhaul_manager_ctx, uint16_t port,
```

Comment:
Shouldn't this be `const backhaul_manager &backhaul_manager_ctx`?

Reply:
Isn't it possible to make the `send_cmdu_to_bus` a friend function to the `agent_ucc_listener` instead of saving the full context?

Reply:
it can't because passing `this` pointer discards the `const` qualifier.
Either way, `friend` won't help since we need the instance of the backhaul manager, which we can get only by passing the 'this' pointer.
`friend` could help if we want to access private members. For example, instead of creating function to get `controller_bridge _mac` as I did in one of the previous commits, it could directly get access to it. I have chosen to create a function for that since that way it could not modify or change anything. This gives more like `const` behavior without actually declaring it as `const`.